nixos/modules/system/boot/stage-2-init.sh

```diff
@@ -165,8 +165,16 @@ exec 1>&$logOutFd 2>&$logErrFd
 exec {logOutFd}>&- {logErrFd}>&-
 
 
+# Create directories for the extra systemd unit paths, so that they can be
+# used. A dangling directory in SYSTEMD_UNIT_PATH gets ignored by systemd.
+for p in @extraSystemdUnitPaths@; do
```

Comment:
Hmm this is the only thing I'm not overly fond of. I don't think we should be creating these arbitrary directories in stage-2.  Especially if people make typos etc.

Reply:
I need to double check this, but what I noticed is that if this directory does not exists, then systemd ignores it. So if after booting this directory gets created, it will not work until the next reboot.

So upgrading from an old systemd release then requires you to reboot to get the `SYSTEMD_UNIT_PATH` setting working, and then another after the directory gets created.